### PR TITLE
[CB-19712] Extend the response with more information

### PR DESCRIPTION
**Pull request by votez**
Added fields :
 1.  _timestamp_ for the last backup date (start date of the backup)
 1. _includedData_ for the backup contents (collection). Possible values: 
      * RANGER_AUDITS
      * RANGER_PERMISSIONS
      * HMS_METADATA
      * ATLAS_METADATA
      * ATLAS_INDEXES

Implementation notes:
* DatalakeRestoreStatusResponse does not extends DatalakeBackupStatusResponse anymore since these are logically different operations. Instead, they both are now extend DatalakeOperationStatus
* Moved isComplete and isFailed to Status enum since it belongs to Status but not Response (left delegate methods for backward compatibility)
* refactoring included : replaced java _assert_ with JUnit _assertXXX_ in tests 

Sample output:
```json
{
    "operationId": "61afae5b-6ab0-4fc4-a8de-9cc30aaf7337",
    "status": "SUCCESSFUL",
    "reason": "No failure messages found",
    "includedData": [
        "RANGER_AUDITS",
        "RANGER_PERMISSIONS",
        "HMS_METADATA",
        "ATLAS_METADATA",
        "ATLAS_INDEXES"
    ],
    "timestamp": "2022-11-22T14:59:56.000895+00:00"
```

tested as:
1) created backup via cdp-cli
1) verified new fields are presented in HTTP request for backup status
1) restored backup via cdp-cli
1) verified status on CDP UI

### CODE REVIEW

File: datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/model/DatalakeBackupStatusResponse.java [hunk 1]
```diff
@@ -1,40 +1,27 @@
...
-    public enum State {
-        STARTED,
-        IN_PROGRESS,
-        SUCCESSFUL,
-        VALIDATION_SUCCESSFUL,
-        VALIDATION_FAILED,
-        FAILED,
-        CANCELLED
-    }
+public class DatalakeBackupStatusResponse implements DatalakeOperationStatus{
```
Comment (RokolyA94) [inline datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/model/DatalakeBackupStatusResponse.java:7]: The build will failed.  (missing space before the `{`)
I suggest that run the `gradle build` and fix all of this `PMD` issues. and push your changes.

File: datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java [hunk 2]
```diff
@@ -18,34 +20,56 @@ public class GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter {
...
+        }
+        if (operationStates.getDatabase().getDatabase().getStatus().equals(SUCCESSFUL_STATUS)) {
+            contents.add("RANGER_PERMISSIONS");
+            contents.add("HMS_METADATA");
+        }
+        if (solrState.getEdgeIndexCollection().getStatus().equals(SUCCESSFUL_STATUS)
+                && solrState.getFulltextIndexCollection().getStatus().equals(SUCCESSFUL_STATUS)
+                && solrState.getVertexIndexCollection().getStatus().equals(SUCCESSFUL_STATUS)
+                && solrState.getEdgeIndexCollection().getStatus().equals(SUCCESSFUL_STATUS)) {
+            contents.add("ATLAS_METADATA");
```
Comment (RokolyA94) [inline datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java:60]: @votez @kkalvagadda1 
`ATLAS_METADATA`, Is It enough? 
Why we do not want to give all of these if we can? 
e.q.:
`EDGE INDEX`
`VERTEX INDEX`
....

**Review by RokolyA94 — APPROVED** [review]
LGTM.

File: datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java (hunk 2 above)
Comment (votez) [reply datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java:62]: according to the Jira ticket the combination of these belongs to Atlas so I only report Atlas not to confuse a client with fine granularity. Would you suggest to add more detailed list of data backed up?

Comment (RokolyA94) [reply datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java:62]: I just asking.
I can accept this.  So I resolved this conversation. Thank you!

File: datalake-api/src/main/java/com/sequenceiq/sdx/api/model/ModelDescriptions.java [hunk 3]
```diff
@@ -195,6 +195,10 @@ public class ModelDescriptions {
 
     public static final String TARGET_MAJOR_VERSION = "The database major version to upgrade to";
 
+    public static final String BACKUP_INCLUDED_DATA = "Data included in the backup (RANGER_PERMISSIONS, HMS_METADATA, RANGER_AUDITS, ATLAS_METADATA)";
```
Comment (kkalvagadda1) [inline datalake-api/src/main/java/com/sequenceiq/sdx/api/model/ModelDescriptions.java:198]: We are considering providing an option for customers to exclude indexes in ATLAS metadata.
Can you break ATLAS_METADATA to ATLAS_METADATA and ATLAS_INDEXES?

File: datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java [hunk 4]
```diff
@@ -18,34 +20,58 @@ public class GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter {
...
+        if (operationStates.getDatabase().getDatabase().getStatus().equals(SUCCESSFUL_STATUS)) {
+            contents.add("RANGER_PERMISSIONS");
+            contents.add("HMS_METADATA");
+        }
+        if (hbaseState.getAtlasEntityAuditEventTable().getStatus().equals(SUCCESSFUL_STATUS)
+                && hbaseState.getAtlasJanusTable().getStatus().equals(SUCCESSFUL_STATUS)
+                && solrState.getFulltextIndexCollection().getStatus().equals(SUCCESSFUL_STATUS)
+                && solrState.getVertexIndexCollection().getStatus().equals(SUCCESSFUL_STATUS)
+                && solrState.getEdgeIndexCollection().getStatus().equals(SUCCESSFUL_STATUS)) {
+            contents.add("ATLAS_METADATA");
```
Comment (kkalvagadda1) [inline datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java:62]: We are considering providing an option for customers to exclude indexes in ATLAS metadata.
Can you break ATLAS_METADATA to ATLAS_METADATA and ATLAS_INDEXES?

if getAtlasJanusTable and getAtlasEntityAuditEventTable are backedup, ATLAS_METADATA is included.
If collections are backedup, ATLAS_INDEXES are included.

**Review by kkalvagadda1 — CHANGES_REQUESTED** [review]
Apart from the few improvements I've requested, over the change looks good.

Your change is extended to the Backup/restore flow as well.

I would recommend you run the below tests
1. Run upgrade validation.
2. run backup/restore from CLI and make sure they are successful and the status rendered in the UI is correct.
3. upgrade/resize if possible.

File: datalake-api/src/main/java/com/sequenceiq/sdx/api/model/ModelDescriptions.java (hunk 3 above)
Comment (votez) [reply datalake-api/src/main/java/com/sequenceiq/sdx/api/model/ModelDescriptions.java]: done

**Review by sxxgrc — APPROVED** [review]

**Review by kkalvagadda1 — APPROVED** [review]
Looks good. Please update the description to capture the improvement you made based on the comments you had. Also, include the additional testing you have done.

**Review by bbihari — APPROVED** [review]

**Pull request merged**